BLUEPRINT/systems/firstwall.py

```diff
@@ -1150,15 +1150,15 @@ def init_equilibrium(self):
         self.x_imp_lcfs = np.min(loop_plane_intersect(self.lcfs, self.mid_plane).T[0])
 
     # Actual run
-    def build(self):
+    def build(self, callback=None):
         """
         Build the 2D profile
         """
         if "profile" in self.inputs:
             self.profile = self.inputs["profile"]
 
-        elif self.inputs.get("FW_optimisation", False):
-            self.profile = self.optimise_fw_profile()
+        elif self.inputs.get("FW_optimisation", False) and callback is not None:
+            callback(self, 0.2, 5)
```

Comment:
Should we delay merging until there is a comment on this?

Reply:
Sorry, I read only now.
There is no explanation. I gave `hf_limit = 0.2` just to make the optimiser do something, otherwise with higher values it wouldn't change the wall profile.
Similarly `n_iteration_max=5`. Just a random value to try if the loop stops after 5 iterations.

Reply:
Ok, @DarioV86 can you make a PR once this PR is in to add a note to the function description (and at the same time move these to default args of the FW_optimisation function).

I'll let @sebkahn have a look at this but I'm happy to merge as is